### ISSUE TITLE
enable CONFIG_DNS_RESOLVER=y

### ISSUE THREAD

**Issue by damoon** [coreos/bugs#1500]
# Issue Report #

### CoreOS Version ###

```
NAME=CoreOS
ID=coreos
VERSION=1068.8.0
VERSION_ID=1068.8.0
BUILD_ID=2016-07-18-0616
PRETTY_NAME="CoreOS 1068.8.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

### Environment ###

vagrant

### Expected Behavior ###

mount cephfs

### Actual Behavior ###

failing to mount cephfs

### Reproduction Steps ###
```
core@core-01 ~ $ sudo mount -t ceph somerandomname.tld:/ /mnt/ -o name=admin,secret=132
mount: wrong fs type, bad option, bad superblock on somerandomname.tld:/,
       missing codepage or helper program, or other error

       In some cases useful info is found in syslog - try
       dmesg | tail or so.
```

### Other Information ###

```
core@core-01 ~ $ dmesg | tail
[  454.887988] SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
[  455.019281] SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
[  455.069510] SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
[  508.416851] SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
[  553.468673] FS-Cache: Loaded
[  553.474534] Key type ceph registered
[  553.475322] libceph: loaded (mon/osd proto 15/24)
[  553.483311] FS-Cache: Netfs 'ceph' registered for caching
[  553.483914] ceph: loaded (mds proto 32)
[  553.484760] libceph: parse_ips bad ip 'somerandomname.tld'
```

### Desired Feature ###

enable CONFIG_DNS_RESOLVER=y

### Other Information ###

i want to use:
ceph-docker for kubernetes:
https://github.com/ceph/ceph-docker/blob/master/examples/kubernetes/ceph-cephfs-test.yaml#L20
ceph monitor ips can't be hard coded here, because they change with every pod redeployment

it uses:
ceph kernel client:
https://github.com/ceph/ceph-client/blob/5ea5c5e0a7f70b256417d3b6e36bd9851504babd/net/ceph/Kconfig#L31
https://github.com/ceph/ceph-client/blob/5ea5c5e0a7f70b256417d3b6e36bd9851504babd/Documentation/networking/dns_resolver.txt
the ceph kernel driver uses CONFIG_DNS_RESOLVER as a default value for CEPH_LIB_USE_DNS_RESOLVER

it is blocked by:
coreos arm kernel:
https://github.com/coreos/coreos-overlay/blob/53cbbbf2653552e8eab527aaca941ab2fc017591/sys-kernel/coreos-modules/files/arm64_defconfig-4.7#L157
the arm version already uses CONFIG_DNS_RESOLVER=y
coreos amd kernel:
https://github.com/coreos/coreos-overlay/blob/53cbbbf2653552e8eab527aaca941ab2fc017591/sys-kernel/coreos-modules/files/amd64_defconfig-4.7
is missing the flag

**Comment by mjg59** [comment]
Will do this and also enable CEPH_FS_POSIX_ACL

**Comment by mischief** [comment]
fixed by coreos/coreos-overlay#2134

**Comment by kayrus** [comment]
@mischief @mjg59 no, it doesn't. you've forgot about `CEPH_FS_POSIX_ACL`

**Comment by mischief** [comment]
ok, will be fixed in coreos/coreos-overlay#2136.
